go/login/server/keymanager.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package backend implements GLOME-login server framework.
+package server
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/google/glome/go/glome"
+)
+
+// ErrInvalidIndex denotes that an invalid index was provided
```

Comment:
There is no indication that this is about keys. I'd use something like `ErrInvalidKeyIndex` rather than just "index" where it's not clear what's happening. Same in the describing comments.

Reply:
SGTM! Name changed in `ErrInvalidIndex` to `ErrInvalideKeyIndex`, and similarly to `ErrOverloadedIndex` and `ErrKeyIndexNotFound`.